src/Razor/src/Microsoft.AspNetCore.Razor.ProjectEngineHost/Utilities/RazorProjectInfoFactory.cs

```diff
@@ -120,21 +121,19 @@ private static RazorConfiguration ComputeRazorConfigurationOptions(AnalyzerConfi
 
         var suppressAddComponentParameter = !compilation.HasAddComponentParameter();
 
-        var razorConfiguration = new RazorConfiguration(
+        var configuration = new RazorConfiguration(
             razorLanguageVersion,
             configurationName,
             Extensions: [],
             UseConsolidatedMvcViews: true,
             suppressAddComponentParameter);
 
-        defaultNamespace = rootNamespace ?? "ASP"; // TODO: Source generator does this. Do we want it?
-
-        return razorConfiguration;
+        return (configuration, rootNamespace ?? "ASP"); // TODO: Source generator does this. Do we want it?
```

Comment:
I think we can just remove the `TODO` from this. It's what we've been doing for a while

Reply:
Or honestly just leave it null

Reply:
> Or honestly just leave it null

This is how subtle bugs creep in! 😛

Reply:
Yeah, I'm not comfortable making an intentional behavior change in this method when I was just making a small refactoring.

Reply:
Fair enough. Can we remove the null check on line 83 at least?

Reply:
and change the return tuple to `(RazorConfiguration configuration, string rootNamespace)`

Reply:
For sure! I had marked `string rootNamespace` as nullable before I even looked at the implementation, since I know that `HostProject.RootNamespace` is nullable.